packages/tools/lib/generate-json-imports/illustrations.js

```diff
@@ -0,0 +1,75 @@
+		const fioriMatches = illustrationName.match(/.*\.js$/);
+		return fioriMatches ? illustrationName : undefined;
+	}).filter(key => !!key);
+
+	const tntDirs = await fs.readdir(tntInputFolder);
+	const tntIllustrationsOnFileSystem = tntDirs.map(illustrationName => {
+		const tntMatches = illustrationName.match(/.*\.js$/);
+		return tntMatches ? illustrationName : undefined;
+	}).filter(key => !!key);
+
+	// dynamic imports for Fiori illustrations
+	const fioriAvailableIllustrationsArray = `[${fioriIllustrationsOnFileSystem.filter(
+		// skipping the items starting with sapIllus-Dialog, sapIllus-Scene, sapIllus-Spot since they are included in the illustration's js file
+		line => !line.startsWith("sapIllus-Dialog") && !line.startsWith("sapIllus-Scene") && !line.startsWith("sapIllus-Spot")).map(illustrationName => `"${illustrationName.replace('.js', '')}"`).join(", ")}]`;
+
+	const fioriDynamicImportLines = fioriIllustrationsOnFileSystem.map(illustrationName =>
+		`\t\tcase "${illustrationName.replace('.js', '')}": return (await import("../../illustrations/${illustrationName}")).default;`).filter(
+			// skipping the items starting with sapIllus-Dialog, sapIllus-Scene, sapIllus-Spot since they are included in the illustration's js file
+			line => !line.startsWith("\t\tcase \"sapIllus-Dialog") && !line.startsWith("\t\tcase \"sapIllus-Scene") && !line.startsWith("\t\tcase \"sapIllus-Spot")).join("\n");
+
+	// dynamic imports for Tnt illustrations
+	const tntAvailableIllustrationsArray = `[${tntIllustrationsOnFileSystem.filter(
+		// skipping the items starting with tnt-Dialog, tnt-Scene, tnt-Spot since they are included in the illustration's js file
+		line => !line.startsWith("tnt-Dialog") && !line.startsWith("tnt-Scene") && !line.startsWith("tnt-Spot")).map(illustrationName => `"${illustrationName.replace('.js', '')}"`).join(", ")}]`;
+
+	const tntDynamicImportLines = tntIllustrationsOnFileSystem.map(illustrationName =>
+		`\t\tcase "Tnt${illustrationName.replace('.js', '')}": return (await import("../../illustrations/tnt/${illustrationName}")).default;`).filter(
+			// skipping the items starting with tnt-Dialog, tnt-Scene, tnt-Spot since they are included in the illustration's js file
+			line => !line.startsWith("\t\tcase \"Tnttnt-Dialog") && !line.startsWith("\t\tcase \"Tnttnt-Scene") && !line.startsWith("\t\tcase \"Tnttnt-Spot")).join("\n");
+
+
+	// dynamic imports file content
+	const contentDynamic = `import { registerIllustrationLoader } from "@ui5/webcomponents-base/dist/asset-registries/Illustrations.js";
+
+const loadIllustration = async (illustrationName) => {
+	switch (illustrationName) {
+${fioriDynamicImportLines}
+${tntDynamicImportLines}
+		default: throw new Error("[Illustrations] Illustration not found: " + illustrationName);
+	}
+};
+	const loadAndCheck = async (illustrationName) => {
+		const data = await loadIllustration(illustrationName);
+		if (illustrationName === "BeforeSearch") {
```

Comment:
I don't quite get the check against `"BeforeSearch"`. 
There is a loader for `"BeforeSearch"` and this is fine. `BeforeSearch` is imported by the IllustratedMessage as default illustration, so it will be already pre-registered and the code will never reach the dynamic import.
So, somehow I miss the purpose of the condition, please elaborate on that.

Reply:
If someone make a typo in the illustration name, this serves as a helper to identify it through the console -
`console.warn("[Illustrations] You are either using the BeforeSearch illustration or you have a typo in the illustration name.");`
Because if you type for example `TntChartArae` - there is a typo, but you still get an illustration even though it is not the right one. 

**In short**: _A helper for finding typos in the illustrations'  names._

Reply:
All right, but still I am not sure about the value of the warning.

Currently, if you have an app (or the IllustratedMessage test page) displaying `"BeforeSearch"` illustration, you will have console warning:
`"[Illustrations] You are either using the "BeforeSearch" illustration or you have a typo in the illustration name."` and warnings should be used when there is something to warn about - using  the "BeforeSearch" illustration is nothing to warn about.

*Note: it happens, because BeforeSearch is not pre-registered - see [comment](https://github.com/SAP/ui5-webcomponents/pull/5927#discussion_r1005415455).
 
In my view, the warning will be useful if it catches and describes the state, when someone sets a wrong `illustrationName` and the component fallbacks to `"BeforeSearch"` as the default value for the `name` property and for example composes a message like:
`[Illustrations]: You've set non-existing  illustration "{my_invalid_name}" , the component fallbacks to the default "BeforeSearch" illustration`.